src/main/java/com/hedvig/productPricing/campaign/service/DomainService.kt

```diff
@@ -297,6 +317,66 @@ class DomainService(
     campaignOwnerRepository.save(campaignOwner)
   }
 
+
+  private fun isReferrerAndRedeemerFromSameMarket(receiverMemberId: String, campaign: Campaign): Boolean {
+
+    // Referrer market
+    val referralCampaign  = campaign as? ReferralCampaign
+    val referrerMemberId = referralCampaign!!.getOwnerAsMember().memberId
```

Comment:
We can leave this method private and do something like

```suggestion
    val referrerMemberId = (referralCampaign.campaignOwner as Member).memberId
```